tests/frontend/test_circuitspecs_chip2.py

```diff
@@ -367,6 +395,62 @@ def test_no_unitary(self, tol):
 
         assert program_equivalence(res, expected, atol=tol)
 
+        # double check that the applied symplectic is correct
+
+        # remove the Fock measurements
+        res.circuit = res.circuit[:-1]
+
+        # extract the Gaussian symplectic matrix
+        O = res.compile("gaussian_unitary").circuit[0].op.p[0]
+
+        # construct the expected symplectic matrix corresponding
+        # to just the initial two mode squeeze gates
+        S = TMS(SQ_AMPLITUDE, 0)
+
+        expected = np.zeros([2*8, 2*8])
+        idx = np.arange(2*8).reshape(4, 4).T
+        for i in idx:
+            expected[i.reshape(-1, 1), i.reshape(1, -1)] = S
+
+        assert np.allclose(O, expected, atol=tol)
+
+    def test_mz_gate(self, tol):
+        """Test that the Mach-Zehnder gate compiles to give the correct unitary"""
+        prog = sf.Program(8)
+
+        with prog.context as q:
+            ops.MZgate(np.pi/2, np.pi) | (q[0], q[1])
+            ops.MZgate(np.pi, 0) | (q[2], q[3])
+            ops.MZgate(np.pi/2, np.pi) | (q[4], q[5])
+            ops.MZgate(np.pi, 0) | (q[6], q[7])
```

Comment:
To avoid sneaky edge cases, might be nice to have a test case at non-exceptional values, e.g., `np.pi/13`?